src/test/java/bio/terra/cda/app/controller/QueryApiControllerTest.java

```diff
@@ -0,0 +1,66 @@
+package bio.terra.cda.app.controller;
+
+import bio.terra.cda.app.service.PingService;
+import bio.terra.cda.app.service.QueryService;
+import bio.terra.cda.generated.model.InlineResponse200;
+import bio.terra.cda.generated.model.Query;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.only;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@Tag("unit")
+@SpringBootTest
+@AutoConfigureMockMvc
+class QueryApiControllerTest {
+
+  @Autowired
+  private MockMvc mvc;
+
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  @MockBean
+  private PingService pingService;
+
+  @MockBean
+  private QueryService queryService;
+
+  private void callQueryApi(boolean dryRun) throws Exception {
+    var query = new Query().nodeType(Query.NodeTypeEnum.COLUMN).value("test");
+    var expected = "SELECT * FROM gdc-bq-sample.cda_mvp.v0 WHERE test";
```

Comment:
This is completely ok for now. We could consider duplicating the table with a name like `.api_unit_test_v0` or something so anyone thinking of deleting the table will think twice.

Reply:
In this case the table is never queried, so it doesn't matter if the table exists or not. This is because the big query service (`QueryService)` is mocked for this test (that is, all methods are no-ops and return null).

It is a problem that this code needs to know the name of the table, though. This should really be using a configurable property, which can be overridden in a test. I'll make an issue for that.